src/main/webapp/app/shared/notification/notification-popup/notification-popup.component.ts

```diff
@@ -96,6 +116,24 @@ export class NotificationPopupComponent implements OnInit {
         }
     }
 
+    /**
+     * Adds a notification about a updated exercise during a live exam to the component's state
+     * and pushes updated problemStatement to student exam exercise via BehaviorSubjects
+     *
+     * @param notification {Notification}
+     */
+    private addExamUpdateNotification(notification: Notification): void {
+        if (notification.target) {
+            const target = JSON.parse(notification.target);
```

Comment:
s.a., I think manually parsing this is not considered good practice. You should be able to use typescript's type mechanisms here.

Reply:
Similar to my response above. I just used the method the legacy code already used in multiple places. I also tried to typecast it but I do not know how to. (`as string` does not do the trick, even though target is string, etc.)